be/src/runtime/memtable_flush_executor.h

```diff
@@ -0,0 +1,85 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <atomic>
+#include <cstdint>
+#include <future>
+#include <memory>
+#include <vector>
+#include <unordered_map>
+#include <utility>
+
+#include "olap/olap_define.h"
+#include "util/blocking_queue.hpp"
+#include "util/spinlock.h"
+#include "util/thread_pool.hpp"
+
+namespace doris {
+
+class ExecEnv;
+class DeltaWriter;
+class MemTable;
+
+// The context for a memtable to be flushed.
+// It does not own any objects in it.
+struct MemTableFlushContext {
+    std::shared_ptr<MemTable> memtable;
+    DeltaWriter* delta_writer;
+    std::atomic<OLAPStatus>* flush_status;
```

Comment:
DeltaWriter will call MemTableFlushExecutor, but you put DeltaWriter in context, which make a loop dependence